unittests/test_typecheck.py

```diff
@@ -33,13 +31,23 @@ def test_bool_type():
     assert types.Bool('True')
     assert types.Bool('YEs')
     assert types.Bool('1')
+    assert types.Bool(True)
     assert not types.Bool('False')
     assert not types.Bool('No')
     assert not types.Bool('0')
+    assert not types.Bool(False)
 
     # Test hash functions
     assert hash(types.Bool('True')) == hash(types.Bool('YES'))
     assert hash(types.Bool('False')) == hash(types.Bool('No'))
+    assert hash(types.Bool(1)) == hash(True)
+    assert hash(types.Bool(0)) == hash(False)
+
+    # Test for equality
+    assert types.Bool(1) == True    # noqa: E712
+    assert True == types.Bool(1)    # noqa: E712
+    assert types.Bool(0) == False   # noqa: E712
+    assert False == types.Bool(0)   # noqa: E712
```

Comment:
Hmm, PEP8 proposes to write such comparisons as `if x is True` or `if x`. The first will not work for bool types, whereas the second one will. I don't know how to treat this actually, except than suggesting to use the second form.

Reply:
But isn't this already used in the tests for boolean conversion?